File: python/02-maximum-pairwise-product/main.py
def maximum_pairwise_product(numbers):
    n = len(numbers)
    first = 0
    second = 0
    for x in range(n):
        if numbers[x] >= first:
            second = first
            first = numbers[x]
        elif numbers[x] > second:
            second = numbers[x]

    return first * second

if __name__ == '__main__':
    arrayNumber = input()
    content = input()
    array = list(map(int, content.split(" ")))
    print(maximum_pairwise_product(array))

File: js/02-maximum-pairwise-product/main.js
// by Alexander Nikolskiy

const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
});

process.stdin.setEncoding('utf8');
rl.once('line', () => {
    rl.on('line', readLine);
});

function readLine (line) {
    const arr = line.toString().split(' ').map(Number);

    console.log(max(arr));
    process.exit();
}

function max(numbers) {
  const n = numbers.length;
  let first = 0;
  let second = 0;

  for (let x = 0; x < n; ++x) {
      if(numbers[x] >= first) {
          second = first;
          first = numbers[x];
      } else if(numbers[x] > second) {
          second = numbers[x];
      }
  }

  return (first * second);
}

module.exports = max;